### PR TITLE
Diffie-Hellman key agreement functionality for Clef

### PR DESCRIPTION
**Pull request by nagydani**
This would allow Clef to be usable not only for signing but also for lookup and decryption of confidential information.

### CODE REVIEW

**Comment by holiman** [comment]
@nagydani I don't understand how this is supposed to work. What is `otherKey` in this context, and what account does the caller wish to utilize? Did you forget to push some commits?

**Comment by holiman** [comment]
Ping @nagydani & @zelig could you shed some light on this?

File: crypto/secp256k1/secp256.go [hunk 1]
```diff
@@ -86,6 +87,22 @@ func Sign(msg []byte, seckey []byte) ([]byte, error) {
 	return sig, nil
 }
 
+// SharedSecret computes the hashed ECDH shared secret.
+// pubkey must be an EC public key
+// seckey must be an EC private key
+func SharedSecret(pubkey []byte, seckey []byte) ([]byte, error) {
```
Comment (holiman) [inline crypto/secp256k1/secp256.go]: I guess this should be added to the wallet-api, to be implemented by `keystore`, `externalsigner` and hardware wallets?

File: signer/core/api.go [hunk 2]
```diff
@@ -430,6 +443,34 @@ func (api *SignerAPI) EcRecover(ctx context.Context, data, sig hexutil.Bytes) (c
...
+	if !res.Approved {
+		return nil, ErrRequestDenied
+	}
+	// Look up the wallet containing the requested signer
+	account := accounts.Account{Address: addr.Address()}
+	wallet, err := api.am.Find(account)
+	if err != nil {
+		return nil, err
+	}
+}
```
Comment (holiman) [inline signer/core/api.go]: So the remainder of this function, should it invoke the `account.SharedSecret( some_addr)` ?

**Comment by fjl** [comment]
Closing because this PR is old and incomplete: the new RPC method doesn't return the shared secret.

**Pull request closed without merging**